tests/platform_tests/api/test_psu.py

```diff
@@ -0,0 +1,124 @@
+                self.expect(isinstance(high_threshold, float), "PSU {} voltage high threshold appears incorrect".format(psu_id))
+            low_threshold = psu.get_voltage_low_threshold(platform_api_conn, psu_id)
+            if self.expect(low_threshold is not None, "Failed to retrieve the low voltage threshold of PSU {}".format(psu_id)):
+                self.expect(isinstance(low_threshold, float), "PSU {} voltage low threshold appears incorrect".format(psu_id))
+            if high_threshold is not None and low_threshold is not None:
+                self.expect(voltage < high_threshold and voltage > low_threshold, "Voltage {} of PSU {} is not in between {} and {}".format(voltage, psu_id, low_threshold, high_threshold))
+        self.assert_expectations()
+
+
+    def test_temperature(self, duthost, localhost, platform_api_conn):
+        ''' PSU temperature test '''
+        for psu_id in range(self.num_psus):
+            temperature = psu.get_temperature(platform_api_conn, psu_id)
+            if self.expect(temperature is not None, "Failed to retrieve temperature of PSU {}".format(psu_id)):
+                self.expect(isinstance(temperature, float), "PSU {} temperature appears incorrect".format(psu_id))
+
+            temp_threshold = psu.get_temperature_high_threshold(platform_api_conn, psu_id)
+            if self.expect(temp_threshold is not None, "Failed to retrieve temperature threshold of PSU {}".format(psu_id)):
+                if self.expect(isinstance(temp_threshold, float), "PSU {} temperature high threshold appears incorrect".format(psu_id)):
+                    self.expect(temperature < temp_threshold, "Temperature {} of PSU {} is over the threshold {}".format(temperature, psu_id, temp_threshold))
+        self.assert_expectations()
+
+
+    def test_led(self, duthost, localhost, platform_api_conn):
+        ''' PSU status led test '''
+        LED_COLOR_LIST = [
+            STATUS_LED_COLOR_GREEN,
+            STATUS_LED_COLOR_AMBER,
+            STATUS_LED_COLOR_RED,
+            STATUS_LED_COLOR_OFF
+        ]
+
+        for psu_id in range(self.num_psus):
+            for color in LED_COLOR_LIST:
+                result = psu.set_status_led(platform_api_conn, psu_id, color)
+                if self.expect(result is not None, "Failed to perform set_status_led of PSU {}".format(psu_id)):
+                    self.expect(result is True, "Failed to set status_led of PSU {}".format(psu_id))
+
+                color_status = psu.get_status_led(platform_api_conn, psu_id)
+                if self.expect(color_status is not None, "Failed to retrieve status_led of PSU {}".format(psu_id)):
+                    if self.expect(isinstance(color_status, str), "PSU {} status led color appears incorrect".format(psu_id)):
+                        self.expect(color == color_status, "Retrived the status_led {} not {} from PSU {}".format(color_status, color, psu_id))
```

Comment:
Suggest to reword message: `"Status LED color incorrect (expected: {}, actual: {}) for PSU {}".format(color, color_status, psu_id))`. Also suggest changing variable names: `color` -> `color_target`, `color_status` -> `color_actual`.

Reply:
I can change color_status to color_actual but will keep the color as it is. it is still readable enough, I think. :)

Reply:
OK. Please also update the message per my suggestion.

Reply:
FYI, the message is still not updated.